### PR TITLE
Specify ObjectStore region via ProviderConfig

### PR DESCRIPTION
**Pull request by tdmanv**
## Change Overview

We cannot discover the region for buckets in AWS S3 that use non-standard regions like us-gov-west-1. This PR adds region to the provider config

This is an alternative to https://github.com/kanisterio/kanister/pull/655

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/objectstore/bucket.go [hunk 1]
```diff
@@ -131,38 +131,40 @@ func (p *provider) ListBuckets(ctx context.Context) (map[string]Bucket, error) {
...
-	var region string
-	if hostEndPoint == "" {
+	region := p.config.Region
+	if region == "" {
 		var err error
 		region, err = p.getRegionForBucket(ctx, bucketName)
 		if err != nil {
 			return nil, errors.Wrapf(err, "could not get region for bucket %s", bucketName)
 		}
+	}
```
Comment (vkamra) [inline pkg/objectstore/bucket.go:162]: Do you need to set `p.config.Region = region` here?

Comment (tdmanv) [reply pkg/objectstore/bucket.go:162]: Yes, thank you. This should probably be hoisted to the provider's initialization so we don't need to do it every time

**Comment by tdmanv** [comment]
Let's not merge til we have the k10 side as well

File: pkg/objectstore/bucket.go [hunk 2]
```diff
@@ -184,8 +186,13 @@ func (p *s3Provider) GetBucket(ctx context.Context, bucketName string) (Bucket,
 }
 
 func (p *s3Provider) DeleteBucket(ctx context.Context, bucketName string) error {
-	region, _ := p.getRegionForBucket(ctx, bucketName)
-	location, err := getStowLocation(ctx, p.config, p.secret, region)
+	cfg := p.config
+	if cfg.Region == "" {
+		// We swalllow this error because region may not be required. If it is,
+		// we'll fail in the next few lines.
+		cfg.Region, _ = p.getRegionForBucket(ctx, bucketName)
```
Comment (DeepikaDixit) [inline pkg/objectstore/bucket.go:193]: We can log this error?

Comment (tdmanv) [reply pkg/objectstore/bucket.go:193]: We should. I"ll followup

**Pull request merged**